Single Product Template - Compatibility Layer: don't skip block without name

gigitux opened the pull request:
This PR fixes the Single Product Compatibility Layer, adding support for custom HTML Blocks. When the Custom HTML is parsed, the result is a block without the name but with the `innerHTML` content.

Currently, there is a check that cleans the blocks without a name, but with this PR, I removed this check.

<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->

<!-- Reference any related issues or PRs here -->

Fixes #9073


### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. Go to Appearance > Editor and in the Single Product template before upgrading to the blockified version, add a HTML Block _outside_ of the Group block as shown in the screenshot below with the contents `<h2>HTML Block here</h2>` (or similar)
2. Save and view this block on the frontend.
3. Return to the template, upgrade it to the blockified version
4. Save and view on the frontend. Be sure that the HTML is visible on the page.


![Screenshot 2023-04-17 at 14 34 13](https://user-images.githubusercontent.com/8639742/232499923-9ca7cb7a-c4e7-417d-af78-a8b86b87ea10.png)


* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog
> Single Product Compatibility Layer: add support for custom HTML Blocks.

gigitux committed:
Single Product Template - Compatibility Layer: don't skip custom HTML

Aljullu reviewed (approved):
Tested and works as expected. Code looks good as well. :shipit: 

Only comment: given that this is a bug that will be fixed, I think it might be worth adding it to the changelog. What do you think?

gigitux commented:
> Tested and works as expected. Code looks good as well. :shipit:
> 
> Only comment: given that this is a bug that will be fixed, I think it might be worth adding it to the changelog. What do you think?

Good point! I added the changelog and removed the label `skip-changelog`. Thanks for the review! 🙇
